src/bloqade/qasm2/dialects/core/stmts.py

```diff
@@ -46,9 +46,9 @@ class Measure(ir.Statement):
 
     name = "measure"
     traits = frozenset({lowering.FromPythonCall()})
-    qarg: ir.SSAValue = info.argument(QubitType)
+    qarg: ir.SSAValue = info.argument(QubitType | QRegType)
     """qarg (Qubit): The qubit to measure."""
-    carg: ir.SSAValue = info.argument(BitType)
+    carg: ir.SSAValue = info.argument(BitType | CRegType)
```

Comment:
you need to add a `def check_type(self)` method to validate the type here because here your definition allows `QubitType` appear with `CRegType`.

Reply:
I added the type check test. FWIW, mixing the types would have already triggered a `RuntimeError` in the impl.

Reply:
It would be better to have the error at compile time compared to run time since in some cases you might not run the function with the simulator for awhile if at all.